File: Programmers/Python/72411.py
counts = {}

def solution(orders, course):
    [findCombis(''.join(sorted(list(order))), '', False, course) for order in orders]
    answer = {c: (2, []) for c in course}
    for key in counts :
        if len(key) in course :
            if answer[len(key)][0] < counts[key] : 
                answer[len(key)] = (counts[key], [key])
            elif answer[len(key)][0] == counts[key] :
                answer[len(key)][1].append(key)
    return sorted(sum([answer[key][1] for key in answer], []))


def findCombis(order, string, passable, courses):
    if (not passable) and len(string) in courses :
        counts[string] = 1 if string not in counts else counts[string]+1
    if len(order) :
        findCombis(order[1:], string, True, courses)
        findCombis(order[1:], string+order[0], False, courses)

File: Programmers/Swift/72411.swift
import Foundation

func solution(_ orders:[String], _ course:[Int]) -> [String] {
    let combinations = courses(orders, course)
    var result: [Int: (Int, [String])] = [:]
    for (combi, count) in combinations {
        let curCount = (result[combi.count] ?? (2, [])).0
        switch count {
        case curCount+1...Int.max : result.updateValue((count, [combi]), forKey: combi.count)
        case curCount : result.updateValue((count, (result[combi.count] ?? (0, [])).1 + [combi]) , forKey: combi.count)
        default : break
        }
    }
    var results: [[String]] = []
    result.forEach { _, combi in
        results.append(combi.1)
    }
    return results.flatMap() {$0}.sorted()
}

func courses(_ orders:[String], _ course: [Int]) -> [String: Int] {
    var combinations: [String: Int] = [:]
    func findCombination(_ order:[Character], _ combination:[Character], _ isContinue: Bool) {
        var order = order
        if !isContinue && course.contains(combination.count) {
            let key = combination.reduce("") {"\($0)\($1)"}
            combinations.updateValue((combinations[key] ?? 0) + 1, forKey: key)
        }
        if !order.isEmpty { 
            let char = order.removeFirst()
            findCombination(order, combination, true)
            findCombination(order, (combination + [char]).sorted(), false)
        }
    }
    orders.forEach() { findCombination(Array($0), [], false) }
    return combinations
}